server/integrations.test.js

```diff
@@ -146,4 +146,25 @@ describe('integration tests', () => {
       server.close(done)
     })
   })
+
+  test(`registerOldPinRemover removes old pins`, async done => {
+    const server = app.listen('9091', async () => {
+      const scheduler = await autoCleanDB(0, 500)
+      const dagRequest = await request(app)
+        .post('/api/v0/dag/put')
+        .send({ test: '123' })
+
+      expect(dagRequest.res.statusCode).toBe(201)
+
+      setTimeout(async () => {
+        const removedPinFile = await Pin.findOne({
+          cid: dagRequest.res.text
+        })
+
+        expect(removedPinFile).toBeNull()
```

Comment:
I think we should add two small things to this test:

1. after we send the `dagRequest`, get back the `cid` of the pinned dag, and check our IPFS node to make sure it's pinned there (optimistic pinning test)
2. after sending the `dagRequest` _then_ call `await autoCleanDB(0, 500)`
3. after checking if the `removedPinFile` is null, check the IPFS node to make sure the `cid` returned in step 1 is no longer pinned

Reply:
Re: 1 and 3... We're testing those things elsewhere, I'm happy to double up but it seems potentially cleaner not to.

Reply:
Where are we testing those behaviors?